File: ecmascript/hamming/hamming.js
function range(n) {
  return [...Array(n).keys()];
}

function zip(a, b) {
  return range(a.length).map(i => [a[i], b[i]]);
}

class Hamming {
  compute(a, b) {
    if (a.length != b.length) {
      throw new Error('DNA strands must be of equal length.');
    }

    return zip(a, b).reduce((acc, [x, y]) => x == y ? acc : acc + 1, 0);
  }
}

export default Hamming;

File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

class Hamming {
  private String a;
  private String b;

  Hamming(String leftStrand, String rightStrand) {
    if (leftStrand.length() != rightStrand.length())
      throw new IllegalArgumentException(
        "leftStrand and rightStrand must be of equal length."
      );

    a = leftStrand;
    b = rightStrand;
  }

  int getHammingDistance() {
    return IntStream.range(0, a.length())
                    .reduce(0, (sum, i) ->
                                 a.charAt(i) == b.charAt(i)
                                 ? sum
                                 : sum + 1);
  }
}
